plugins/pulp_rpm/plugins/db/models.py

```diff
@@ -1190,6 +1198,26 @@ def get_unique_pkglists(cls, errata_id):
                                                     batchSize=5).next()['pkglists']
         return pkglists
 
+    @classmethod
+    def get_unique_modules(cls, errata_id):
+        """
+        Generate a list of unique module search dicsts for a specified erratum.
+
+        :param errata_id: The erratum to generate a unique set of modules for
+        :type errata_id: str
+        :return: unique modules for a specified erratum
+        :rtype: a lists of dicts
+        """
+        match_stage = {'$match': {'errata_id': errata_id, 'collections.module': {'$exists': True}}}
+        project_stage = {'$project': {'collections.module': True}}
+        unwind_stage = {'$unwind': '$collections'}
+        group_stage = {'$group': {'_id': '$errata_id',
```

Comment:
Can you elaborate why it's not enough to have just a match stage and a group stage? I probably miss something, mongo data manipulation is often counter-intuitive :)

Reply:
`match` reduces the units processed and `project` minimizes the data amount processed. Just the `module` data needs to be picked. `unwind` multiplies the amount of units by popping them out of the nested lists --- this is cool thanks to the `project` stage. As a result, the `group` stage returns a flat structure. Can rework the same as the `pkglists` equivalent tho.